### PR TITLE
Only download legit songs

### PR DESCRIPTION
**Pull request by davepagurek**
Before downloading a song, this checks that:
- The `content-type` header is `audio/something`
- The file is more than 2mb to avoid bad quality downloads and 30-second samples. The threshold can be changed in `MIN_MB` in `player.js`.

When a song is downloading, it sets a downloading variable, and clears it when it's done. The auto-updating is done like this:
- Create a div with an id to for the content to be auto-updated
- Make an endpoint that just renders html for that div
- add `{{auto_update "id" "/some/endpoint" seconds}}` to the bottom of a view

### CODE REVIEW

File: public/javascripts/helpers.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+function ajax(method, url, callback) {
+    var xmlhttp;
+
+    if (window.XMLHttpRequest) {
+        // code for IE7+, Firefox, Chrome, Opera, Safari
+        xmlhttp = new XMLHttpRequest();
+    } else {
+        // code for IE6, IE5
+        xmlhttp = new ActiveXObject("Microsoft.XMLHTTP");
```
Comment (andrew749) [inline public/javascripts/helpers.js]: do we really need to support ie5 and 6. like we r gonna run this on somethign with webkit or firefox

Comment (andrew749) [inline public/javascripts/helpers.js]: Actually i guess its good practice.

Comment (davepagurek) [inline public/javascripts/helpers.js]: This was copy-and-pasted from an earlier project that needed support, I can definitely take this out

File: plugins/player.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
 module.exports = function(pluto) {
+    var MIN_MB = 3;
```
Comment (andrew749) [inline plugins/player.js:2]: we should probably adjust this to 2 or 1.5

File: plugins/player.js [hunk 3]
```diff
@@ -21,26 +26,29 @@ module.exports = function(pluto) {
...
-                    } else {
-                        console.log("error trying another link");
+                request.head(url, function(err, res, body) {
+                    if (err || res.headers['content-type'].indexOf("audio") == -1 || res.headers['content-length']/1000000 < MIN_MB) {
                         songs[song.id].ignore.push(url);
                         pluto.saveStorage("songs");
                         pluto.emitEvent("music::play", song);
+                    } else {
+                        downloading.song = song;
+                        request(url).pipe(fs.createWriteStream(songURL)).on('close', function() {
```
Comment (davepagurek) [inline plugins/player.js]: Going to get progress from this https://github.com/IndigoUnited/node-request-progress

**Comment by davepagurek** [comment]
Now includes percent!
screen shot 2015-08-25 at 10 10 36 pm

**Pull request merged**